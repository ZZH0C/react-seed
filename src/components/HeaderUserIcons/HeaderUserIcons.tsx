import React from 'react';
// import '../../styles/main.css';

interface HeaderUserIconsProps {
  classname: 'icon-mail' | 'icon-bell';
}

export const HeaderUserIcons: React.FC<HeaderUserIconsProps> = ({
  classname,
}) => {
  return (
    <a href="index.html" className="btn btn-sm btn-header">
      <i className={`headerIcon ${classname}`} />
    </a>
  );
};
