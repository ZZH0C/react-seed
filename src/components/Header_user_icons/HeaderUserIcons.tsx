import React from 'react';
// import '../../styles/main.css';

interface HeaderUserIconsProps {
  style: 'headerIcon icon-mail' | 'headerIcon icon-bell';
}

export const HeaderUserIcons: React.FC<HeaderUserIconsProps> = ({ style }) => {
  return (
    <a href="index.html" className="btn btn-sm btn-header">
      <i className={style} />
    </a>
  );
};
