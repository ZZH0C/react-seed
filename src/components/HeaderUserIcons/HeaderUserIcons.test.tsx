import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { HeaderUserIcons } from './HeaderUserIcons';

describe('components/Head', () => {
  const mockProps = {
    mockURL: 'mockURL',
  };
  const firstClass = 'icon-mail';
  const secondClass = 'icon-bell';

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render and match snapshot with first classname', () => {
    expect(
      shallow(
        <HeaderUserIcons url={mockProps.mockURL} classname={firstClass} />,
      ),
    ).toMatchSnapshot();
  });
  it('should render and match snapshot with second classname', () => {
    expect(
      shallow(
        <HeaderUserIcons url={mockProps.mockURL} classname={secondClass} />,
      ),
    ).toMatchSnapshot();
  });
});
