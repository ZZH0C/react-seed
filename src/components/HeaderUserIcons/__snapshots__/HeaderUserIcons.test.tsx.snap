// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Head should render and match snapshot with first classname 1`] = `
<a
  className="btn btn-sm btn-header"
  href="mockURL"
>
  <i
    className="headerIcon icon-mail"
  />
</a>
`;

exports[`components/Head should render and match snapshot with second classname 1`] = `
<a
  className="btn btn-sm btn-header"
  href="mockURL"
>
  <i
    className="headerIcon icon-bell"
  />
</a>
`;
